pkg/api/server.go

```diff
@@ -52,7 +52,7 @@ func NewServer(
 	port int,
 	store storage.Store,
 	asyncEngine async.Engine,
-	authenticator authenticator.Authenticator,
+	filterChain filters.Chain,
```

Comment:
The server was essentially applying the authenticator as a filter previously. Now we allow a chain of them to be passed to the server. This could be just the authenticator. It could be multiple authenticators later on (like if we add support for opaque bearer tokens). This PR also adds in a header validator.